### PR TITLE
[yang] syslog yang model

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshmi Narasimhan <arlakshm@microsoft.com>

#### Why I did it
Add yang model for syslog server
#### How I did it
Add new file `sonic-syslog.yang` and new files for tests

#### How to verify it
Compile `target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl`

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

**Review by renukamanavalan — DISMISSED** [review]

File: src/sonic-yang-models/yang-models/sonic-syslog.yang [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    container sonic-syslog {
+
+        container SYSLOG_SERVER {
+
+            description "SYSLOG SERVER part of config_db.json";
+
+            list SYSLOG_SERVER_LIST {
+                max-elements 10;
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-syslog.yang:19]: Where does 10 come from?

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            "sonic-syslog:SYSLOG_SERVER": {
+                "SYSLOG_SERVER_LIST": [
+                    {
+                        "server_address": ""
+                    }
+                ]
+            }
+        }
+    },
+    "SYSLOG_SERVER_2_NEG_TEST": {
```
Comment (wen587) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json:27]: Hi, could you add a NEG_TEST for invalid IPv4 address such as `1000.0.0.5`
I tested locally but I don't know why it doesn't block invalid IPv4 address.

**Comment by wen587** [comment]
Hi, could you add a NEG_TEST for invalid IPv4 address such as 1000.0.0.5
I tested locally but I don't know why it doesn't block invalid IPv4 address.

File: src/sonic-yang-models/yang-models/sonic-syslog.yang (hunk 1 above)
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-syslog.yang:19]: Remove the max-elements

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json (hunk 2 above)
Comment (arlakshm) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json:27]: address in the latest commit.

File: src/sonic-yang-models/yang-models/sonic-syslog.yang (hunk 1 above)
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-syslog.yang]: address in latest commit

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                ]
+            }
+        }
+    },
+    "SYSLOG_SERVER_INVALID_IPADDR_TEST" : {
+        "sonic-syslog:sonic-syslog": {
+            "sonic-syslog:SYSLOG_SERVER": {
+                "SYSLOG_SERVER_LIST": [
+                    {
+                        "server_address": "1111:22:33.1"
```
Comment (wen587) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json:43]: Hi arlakshm, could you change it to an invalid IPv4 address? Like `1111.22.33.1`
From my local test, invalid IPv4 address will not be blocked.

Comment (wen587) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json:43]: From my [Syslog server test#4725](https://github.com/Azure/sonic-mgmt/pull/4725), the invalid IPv4(test_syslog_server_tc3_xfail) was expected to fail but succeed.
```
def test_syslog_server_tc3_xfail(duthost, setup_env, op,
        dummy_syslog_server_v4, dummy_syslog_server_v6):
    """ Test expect fail testcase
    ("add", "10.0.0.587", "cc98:2008::1"), ADD Invalid IPv4 address
```

File: src/sonic-yang-models/yang-models/sonic-syslog.yang (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-syslog.yang]: Also remove in the description ", limiting to 10" ?

**Comment by arlakshm** [comment]
> Hi, could you add a NEG_TEST for invalid IPv4 address such as 1000.0.0.5 I tested locally but I don't know why it doesn't block invalid IPv4 address.

@wen587, I have updated the constraint and test in the latest commit.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json (hunk 3 above)
Comment (arlakshm) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/syslog.json]: address in the latest commit

File: src/sonic-yang-models/yang-models/sonic-syslog.yang (hunk 1 above)
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-syslog.yang]: removed the description in the latest commit

**Comment by wen587** [comment]
Hi @arlakshm , thanks for the update.
Is it possible to support ipv6 syslog server in yang? Seems `config syslog add` does support ipv6 server.

**Comment by arlakshm** [comment]
> Seems `config syslog add` does support ipv6 server.

Hi @wen587, as far as I know we support only ipv4 syslog server.  The CLI command might be missing the IP address validation ?
@qiluo-msft , @renukamanavalan can you confirm

**Comment by qiluo-msft** [comment]
> > Seems `config syslog add` does support ipv6 server.
> 
> Hi @wen587, as far as I know we support only ipv4 syslog server. The CLI command might be missing the IP address validation ? @qiluo-msft , @renukamanavalan can you confirm

Checking https://github.com/Azure/sonic-buildimage/pull/3754, we should support ipv6 syslog server address. Otherwise it will be a bug.

**Comment by arlakshm** [comment]
> > > Seems `config syslog add` does support ipv6 server.
> > 
> > 
> > Hi @wen587, as far as I know we support only ipv4 syslog server. The CLI command might be missing the IP address validation ? @qiluo-msft , @renukamanavalan can you confirm
> 
> Checking #3754, we should support ipv6 syslog server address. Otherwise it will be a bug.

Add support of ipv6 syslog server in the latest commit

**Review by qiluo-msft — APPROVED** [review]
LGTM. Please also check with other reviewers for active comments.

**Review by wen587 — APPROVED** [review]
lgtm. Thanks.

**Review by praveen-li — APPROVED** [review]
LGTM

**Pull request merged**